core/src/test/java/kafka/test/MockReplicaSelector.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kafka.test;
+
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.replica.ClientMetadata;
+import org.apache.kafka.common.replica.PartitionView;
+import org.apache.kafka.common.replica.ReplicaSelector;
+import org.apache.kafka.common.replica.ReplicaView;
+
+import java.util.Optional;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class MockReplicaSelector implements ReplicaSelector {
```

Comment:
I am not a fan of putting it in this package. Out of curiosity, have you tried to define it in the test itself? I guess that should work.